app/models/user.rb

```diff
@@ -22,7 +22,7 @@ class User < ApplicationRecord
 
   before_validation {
     self.username = username.downcase
-    self.email = email.downcase
+    self.email = email ? email.downcase : "#{username}@example.com"
```

Comment:
There's gonna be a merge conflict here, but I'm a bit shaky on using `example.com`. I did something like this:

`self.username = (username.to_s == '') ? username : username.downcase`
`self.email = (email.to_s == '') ? email : email.downcase`